pipeline/stimulation.py

```diff
@@ -37,3 +38,67 @@ class PhotoStimulationInfo(dj.Manual):
    photo_stim_notes = '':                varchar(128)
    """  
 
+
+@schema
+class PhotoStimulation(dj.Manual):
+    definition = """  #  Photostimulus profile used for stimulation in this session
+    -> acquisition.Session
+    photostim_datetime: datetime # the time of performing this stimulation with respect to start time of the session, in the scenario of multiple stimulations per session
+    ---
+    -> PhotoStimulationInfo
+    photostim_timeseries=null: longblob  # (mW)
+    photostim_start_time=null: float  # (s) first timepoint of photostim recording
+    photostim_sampling_rate=null: float  # (Hz) sampling rate of photostim recording
+    """
+
+
+@schema
+class TrialPhotoStimInfo(dj.Imported):
+    definition = """ # information related to the stimulation settings for this trial
+    -> acquisition.TrialSet.Trial
+    ---
+    photo_stim_type: enum('stimulation','inhibition','N/A')
+    photo_stim_period: enum('sample','delay','response','N/A')
+    photo_stim_power: float  # (mW) stimulation power 
+    photo_loc_galvo_x: float  # (mm) photostim coordinates field 
+    photo_loc_galvo_y: float  # (mm) photostim coordinates field 
+    photo_loc_galvo_z: float  # (mm) photostim coordinates field 
+    """
+
+    def make(self, key):
+        # this function implements the ingestion of Trial stim info into the pipeline
+        return None
+
+
+@schema
+class TrialSegmentedPhotoStimulus(dj.Computed):
+    definition = """
+    -> PhotoStimulation
+    -> acquisition.TrialSet.Trial
+    -> analysis.TrialSegmentationSetting
+    ---
+    segmented_photostim: longblob
+    """
+
+    # custom key_source where acquisition.PhotoStimulation.photostim_timeseries exist
+    key_source = acquisition.TrialSet.Trial * analysis.TrialSegmentationSetting * (
+                PhotoStimulation - 'photostim_timeseries is NULL')
+
+    def make(self, key):
+        # get event, pre/post stim duration
+        event_name, pre_stim_dur, post_stim_dur = (analysis.TrialSegmentationSetting & key).fetch1(
+            'event', 'pre_stim_duration', 'post_stim_duration')
+        # get raw
+        fs, first_time_point, photostim_timeseries = (PhotoStimulation & key).fetch1(
+            'photostim_sampling_rate', 'photostim_start_time', 'photostim_timeseries')
+        # segmentation
+        try:
+            key['segmented_photostim'] = analysis.perform_trial_segmentation(key, event_name, pre_stim_dur,
+                                                                             post_stim_dur, photostim_timeseries,
+                                                                             fs, first_time_point)
+        except analysis.EventChoiceError as e:
+            print(f'Trial segmentation error - Msg: {str(e)}')
```

Comment:
```suggestion
            raise PipelineError(f'Trial segmentation error - Msg: {str(e)}')
```
Do not suppress errors from `make`.  Errors are handled by the framework in a systematic way.